library/model/book.py

```diff
@@ -96,8 +62,10 @@ def borrow_book(self) -> "BorrowedBook":
 
     def __eq__(self, other):
         """Overrides the default implementation"""
-        if isinstance(other, Book) or isinstance(other, BorrowedBook):
-            return self.isbn == other.isbn and self._book_type == other._book_type
+        if isinstance(other, Book):
+            return self.isbn == other.isbn and isinstance(self, type(other))
```

Comment:
Would expect `type(self) is type(other)`, because that way you only check if self is a subclass (or same class as ) of other.